src/mscorlib/src/System/Buffer.cs

```diff
@@ -618,5 +455,13 @@ public static unsafe void MemoryCopy(void* source, void* destination, ulong dest
             Memmove((byte*)destination, (byte*)source, checked((uint)sourceBytesToCopy));
 #endif // BIT64
         }
+        
+#if BIT64        
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        private struct Block16 { }
+
+        [StructLayout(LayoutKind.Sequential, Size = 64)]
+        private struct Block64 { } 
```

Comment:
x86 CoreCLR switched to RyuJIT so there shouldn't be any differences between X86 and X64 here, both should use `Block64`.

I don't know what ARM64 does but it would make sense for it to use the best available method for moving a 64 bytes block, that should be the same or better than moving 4 bytes at a time.

It may be better to handle this by adding some specific define at the top of the file so it's easy to change:
```C#
#if AMD64 || X86
#define HAS_BLOCK64
#endif
```